src/FunctionAppResolver.ts

```diff
@@ -10,9 +10,9 @@ export class FunctionAppResolver implements AppResourceResolver {
             const client = await createWebSiteClient({ ...context, ...subContext });
             const rg = getResourceGroupFromId(nonNullProp(resource, 'id'));
             const name = nonNullProp(resource, 'name');
+
             const site = await client.webApps.get(rg, name);
-            site.siteConfig = await client.webApps.getConfiguration(rg, name)
-            return new ResolvedFunctionAppResource(subContext, site);
+            return ResolvedFunctionAppResource.createResolvedFunctionAppResource(context, subContext, site);
```

Comment:
Not sure I understand the function of the latest commit (where `siteConfig` is loaded async).

Reply:
I wanted to make it lazy, but it would require changes in `viewProperties` in Resource Groups. Unless I can make a get/set data asynchronous, but I don't think that's possible.

Loading the siteConfig in the background is to not slow down the resolve, but still having it likely by the time a user views the properties.

Reply:
Does part of the `siteConfig` end up being displayed?

Reply:
Yeah, I think that's the way to go (change `viewProperties`). But I think it probably should be done in a separate PR.